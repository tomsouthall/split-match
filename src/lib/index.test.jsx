import React from 'react'
import renderer from 'react-test-renderer'

import { describe, expect, test } from 'vitest'
import SplitMatch from './index'

const strNYC = 'New York City, New York, United States'
const strNewNewNew = 'New, New, New'

const SplitComponent = (props) => {
  const { children, index } = props
  const className = `split${index}`
  return <span className={className}>{children}</span>
}

const MatchComponent = (props) => {
  const { children } = props
  return <span className="match">{children}</span>
}

describe('SplitMatch', () => {
  test('renders correctly when there is nothing to split or match', () => {
    const component = renderer.create(
      <SplitMatch>foobar</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly when only splitting', () => {
    const component = renderer.create(
      <SplitMatch>{strNYC}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly when splitting and matching', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={false}
        globalSplit={false}
        includeSeparator={true}
        searchText={'New York City, New'}
        separator=",">{strNYC}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('splits first only', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={false}
        globalSplit={false}
        includeSeparator={true}
        searchText={'New'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('splits globally', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={false}
        globalSplit={true}
        includeSeparator={true}
        searchText={'New'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('matches first only', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={false}
        globalSplit={true}
        includeSeparator={true}
        searchText={'New'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('matches globally', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={true}
        searchText={'New'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('matches across splits', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={true}
        searchText={'w, N'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('does not split if cases do not match', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={true}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={true}
        searchText={'New'}
        separator="n">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('splits if cases do match', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={true}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={true}
        searchText={'New'}
        separator="N">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('does not match if cases do not match', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={true}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={false}
        includeSeparator={true}
        searchText={'new'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('removes all separators if globalSplit', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={false}
        searchText={'new'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('removes first separator if not globalSplit', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={false}
        includeSeparator={false}
        searchText={'new'}
        separator=",">{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('uses SplitComponent when supplied as prop', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={false}
        searchText={'new'}
        separator=","
        SplitComponent={SplitComponent}>{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('uses MatchComponent when supplied as prop', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={false}
        searchText={'new'}
        separator=","
        MatchComponent={MatchComponent}>{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('uses both SplitComponent and MatchComponent when supplied as props', () => {
    const component = renderer.create(
      <SplitMatch
        caseSensitiveMatch={false}
        caseSensitiveSplit={false}
        globalMatch={true}
        globalSplit={true}
        includeSeparator={false}
        searchText={'new'}
        separator=","
        SplitComponent={SplitComponent}
        MatchComponent={MatchComponent}>{strNewNewNew}</SplitMatch>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})