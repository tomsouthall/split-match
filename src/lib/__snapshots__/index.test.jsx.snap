// Vitest Snapshot v1

exports[`SplitMatch > does not match if cases do not match 1`] = `
[
  <span>
    New,
  </span>,
  <span>
     New, New
  </span>,
]
`;

exports[`SplitMatch > does not split if cases do not match 1`] = `
<span>
  <strong>
    New
  </strong>
  , 
  <strong>
    New
  </strong>
  , 
  <strong>
    New
  </strong>
</span>
`;

exports[`SplitMatch > matches across splits 1`] = `
[
  <span>
    Ne
    <strong>
      w,
    </strong>
  </span>,
  <span>
    <strong>
       N
    </strong>
    e
    <strong>
      w,
    </strong>
  </span>,
  <span>
    <strong>
       N
    </strong>
    ew
  </span>,
]
`;

exports[`SplitMatch > matches first only 1`] = `
[
  <span>
    <strong>
      New
    </strong>
    ,
  </span>,
  <span>
     New,
  </span>,
  <span>
     New
  </span>,
]
`;

exports[`SplitMatch > matches globally 1`] = `
[
  <span>
    <strong>
      New
    </strong>
    ,
  </span>,
  <span>
     
    <strong>
      New
    </strong>
    ,
  </span>,
  <span>
     
    <strong>
      New
    </strong>
  </span>,
]
`;

exports[`SplitMatch > removes all separators if globalSplit 1`] = `
[
  <span>
    <strong>
      New
    </strong>
    
  </span>,
  <span>
     
    <strong>
      New
    </strong>
    
  </span>,
  <span>
     
    <strong>
      New
    </strong>
  </span>,
]
`;

exports[`SplitMatch > removes first separator if not globalSplit 1`] = `
[
  <span>
    <strong>
      New
    </strong>
    
  </span>,
  <span>
     
    <strong>
      New
    </strong>
    , 
    <strong>
      New
    </strong>
  </span>,
]
`;

exports[`SplitMatch > renders correctly when only splitting 1`] = `
[
  <span>
    New York City,
  </span>,
  <span>
     New York,
  </span>,
  <span>
     United States
  </span>,
]
`;

exports[`SplitMatch > renders correctly when splitting and matching 1`] = `
[
  <span>
    <strong>
      New York City,
    </strong>
  </span>,
  <span>
    <strong>
       New
    </strong>
     York, United States
  </span>,
]
`;

exports[`SplitMatch > renders correctly when there is nothing to split or match 1`] = `
<span>
  foobar
</span>
`;

exports[`SplitMatch > splits first only 1`] = `
[
  <span>
    <strong>
      New
    </strong>
    ,
  </span>,
  <span>
     New, New
  </span>,
]
`;

exports[`SplitMatch > splits globally 1`] = `
[
  <span>
    <strong>
      New
    </strong>
    ,
  </span>,
  <span>
     New,
  </span>,
  <span>
     New
  </span>,
]
`;

exports[`SplitMatch > splits if cases do match 1`] = `
[
  <span>
    <strong>
      N
    </strong>
  </span>,
  <span>
    <strong>
      ew
    </strong>
    , 
    <strong>
      N
    </strong>
  </span>,
  <span>
    <strong>
      ew
    </strong>
    , 
    <strong>
      N
    </strong>
  </span>,
  <span>
    <strong>
      ew
    </strong>
  </span>,
]
`;

exports[`SplitMatch > uses MatchComponent when supplied as prop 1`] = `
[
  <span>
    <span
      className="match"
    >
      New
    </span>
    
  </span>,
  <span>
     
    <span
      className="match"
    >
      New
    </span>
    
  </span>,
  <span>
     
    <span
      className="match"
    >
      New
    </span>
  </span>,
]
`;

exports[`SplitMatch > uses SplitComponent when supplied as prop 1`] = `
[
  <span
    className="split0"
  >
    <strong>
      New
    </strong>
    
  </span>,
  <span
    className="split1"
  >
     
    <strong>
      New
    </strong>
    
  </span>,
  <span
    className="split2"
  >
     
    <strong>
      New
    </strong>
  </span>,
]
`;

exports[`SplitMatch > uses both SplitComponent and MatchComponent when supplied as props 1`] = `
[
  <span
    className="split0"
  >
    <span
      className="match"
    >
      New
    </span>
    
  </span>,
  <span
    className="split1"
  >
     
    <span
      className="match"
    >
      New
    </span>
    
  </span>,
  <span
    className="split2"
  >
     
    <span
      className="match"
    >
      New
    </span>
  </span>,
]
`;
